streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -262,7 +262,7 @@ public void close() {
         if (eosEnabled) {
             streamsProducer.abortTransaction();
         }
-        streamsProducer.close();
```

Comment:
Only the creator may close the producer, but we can go ahead and call `flush()` here.